### PR TITLE
Supply tracker (Food, ammo, trident casts, and teleport tabs/scrolls)

### PR DESCRIPTION
**Pull request by DWhite72**
Automatically keeps track of all consumables, ammo, trident casts, teleport tablets, and teleport scrolls used during the session and allows user to add, edit, or delete the amount.(Additions are right click submenu, Amount edit is pulls up a new gui element)
Works for all consumables, ammo, throwing weapons(including chins), trident casts(seas, swamp, sang, and their (e) versions), teleport scrolls, and teleport tablets.
Works with tick eating.
Built in protection against spam clicking adding too many of one item.
Arrow tracking works even if ammo type is switched or more of the same type is added.

[Main Features](https://i.gyazo.com/bd1d02ac3522de673ce0a373be9f52e0.mp4)
[Longer session test](https://i.gyazo.com/0d2e3a5dce47a7dedc6193f08ab3929f.png)
[Arrow tracking](https://i.gyazo.com/a9ae578f414005268af5fbd264bb2fd7.mp4)
[Throwing weapon Tracking](https://i.gyazo.com/1cd2ac25dd03bb1cf0d19694b6ff37f4.mp4)
[Add items from sub-menu](https://i.gyazo.com/f955f90cada806ff54a0341b1b3febc2.mp4)
[10 trip Zulrah test](https://i.gyazo.com/31492212d278f356502ceb7bf031e0a8.png)

### CODE REVIEW

**Comment by Kyle-Newton** [comment]
I remember seeing this last week in Reddit, it's amazing and thanks for all the work you put into it :) Hope to see it merged soon!

**Comment by psikoi** [comment]
You should try to make this plugin fit more into the client's design style, pm me on discord and I can give you some tips on it

**Comment by DWhite72** [comment]
Will do!

**Comment by Kyle-Newton** [comment]
Also was thinking about this, does it happen to handle things like Pot Share yet? Where it would cost both the runes and the X number of sips used?

**Comment by DWhite72** [comment]
At the moment it doesn't track spells hoping to add that later. The only way to track something like that would be to right click the item and edit the amount

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+	@Inject
+	private Client client;
+
+	
+	@Override
+	protected void startUp() throws Exception
+	{
+		panel = new SuppliesTrackerPanel(itemManager);
+		suppliesTrackerPlugin_instance = this;
+		final BufferedImage header = ImageUtil.getResourceStreamFromClass(getClass(), "sarabrew.png");
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java]: I believe the size of the image was the issue so i got a different one and cleaned it up a bit to look better while increasing size and yeah jpg needs to be removed I'll change that

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+	void buildEntries(int itemId) throws ExecutionException
+	{
+			final ItemComposition itemComposition = itemManager.getItemComposition(itemId);
+			String name = itemComposition.getName();
+			long price;
+			boolean potion = false;
+			boolean pizzapie = false;
+
+			//Ignore COX potions
+			if (name.toLowerCase().contains("xeric's") || name.toLowerCase().contains("elder") || name.toLowerCase().contains("twisted") || name.toLowerCase().contains("revitalisation") || name.toLowerCase().contains("overload") || name.toLowerCase().contains("prayer enhance"))
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java]: Fixed in next pull

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+			{
+				return;
+			}
+			//Ignore COX fish
+			if (name.toLowerCase().contains("pysk") || name.toLowerCase().contains("suphi") || name.toLowerCase().contains("leckish") || name.toLowerCase().contains("brawk") || name.toLowerCase().contains("mycil") || name.toLowerCase().contains("roqed") || name.toLowerCase().contains("kyren"))
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java]: Fixed in next pull

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java [hunk 4]
```diff
@@ -0,0 +1,626 @@
...
+			{
+				return;
+			}
+			//Ignore COX bats
+			if (name.toLowerCase().contains("guanic") || name.toLowerCase().contains("prael") || name.toLowerCase().contains("giral") || name.toLowerCase().contains("phluxia") || name.toLowerCase().contains("kryket") || name.toLowerCase().contains("murng") || name.toLowerCase().contains("psykk"))
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java]: Fixed in next pull

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java [hunk 5]
```diff
@@ -0,0 +1,626 @@
...
+	private ItemContainer old;
+	private Deque<Integer> itemStack = new ArrayDeque<>();
+	private Deque<Integer> slotStack = new ArrayDeque<>();
+	private Deque<String> typeStack = new ArrayDeque<>();
+	private Deque<net.runelite.api.Item[]> oldInventStack = new ArrayDeque<>();
+	private int ammoId = 0;
+	private int ammoAmount = 0;
+	private int thrownId = 0;
+	private int thrownAmount = 0;
+	private int[] throwingIds = new int[]{806, 807, 808, 3093, 809, 810, 811, 11230, 864, 863, 865, 869, 866, 867, 868, 800, 801, 802, 803, 804, 805, 20849, 10033, 10034, 11959};
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPlugin.java]: throwingIds are the itemid of throwing weapons to check if the player has a throwing weapon equipped

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java [hunk 6]
```diff
@@ -0,0 +1,2148 @@
...
+				break;
+		}
+		return itemId;
+	}
+
+	private int getHalf(int itemId)
+	{
+		switch (itemId)
+		{
+			case 2297:
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java]: Switches full pizza and pie ids to get the image for half. Added comments in file

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java [hunk 7]
```diff
@@ -0,0 +1,2148 @@
...
+			}
+		};
+		setAmountRow.amount.addKeyListener(keyAdapter);
+	}
+
+	private int getSlice(int itemId)
+	{
+		switch (itemId)
+		{
+			case 1891:
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java]: Switches full cake ids to get the image for slice. Added comments in file

[the remark this answers is not in the document]
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java]: Why not just use the defined item ID instead? That way no comments are needed.

File: runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java [hunk 8]
```diff
@@ -0,0 +1,2148 @@
...
+		final JMenuItem redChinchompa = new JMenuItem("Red Chinchompa");
+		final JMenuItem blackChinchompa = new JMenuItem("Black Chinchompa");
+
+		//Add menu options
+		//Food
+		addFood.add(lobster).addActionListener(e ->
+		{
+			try
+			{
+				SuppliesTrackerPlugin.getInstance().buildEntries(379);
```
[the remark this answers is not in the document]
Comment (DWhite72) [reply runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java]: Id corresponding to the item clicked on. Example below the addRune menu Lava rune choice adds a lava rune(4699)

```java
"addRune.add(lava).addActionListener(e ->
		{
			try
			{
				SuppliesTrackerPlugin.getInstance().buildEntries(4699);
			}
			catch
					(ExecutionException e1)
			{
				e1.printStackTrace();
			}
		});"
```

**Comment by psikoi** [comment]
What happened?

**Comment by luisferrassini** [comment]
What happened?

**Pull request closed without merging**